### DISCUSSION TITLE
Is there anywhere I can easily confirm if this library counts as a munition?

### DISCUSSION THREAD

**Discussion by cowens** [openpgpjs/openpgpjs#1217] [Q&A]
I am worried about US laws concern export of encryption algorithms that are considered munitions.  From what I can glean from [this EFF article about US Export Controls](https://www.eff.org/deeplinks/2019/08/us-export-controls-and-published-encryption-source-code-explained), I don't have to worry about algorithms that have been "published" where "published" means the URL for the source code (or copies of the source code) has been sent to crypt@bis.doc.gov and enc@nsa.gov for each change.

Does this project handle that (via, say, a git hook) or is this an effort I need to take on for myself?

**Comment by bartbutler** [comment]
I think you can safely put that worry to rest. This library is an implementation of very standard cryptographic algorithms codified in RFC4880 and RFC4880-bis, and there's nothing particularly novel here cryptographically. Also, the devs are an international group so it's hard to see how US would have jurisdiction in any case, but I understand the desire to be "safe."

**Comment by cowens** [comment]
The US doesn't have jurisdiction over the international developers, but it does have jurisdiction over US companies that export the code the international developers wrote. As an analogy, A French company makes a product which is purchased by a US company and shipped to a warehouse in the US.  The product is then sold and shipped to a person in England.  This would count as reexport (IMHO/IANAL) and reexport of encryption source code is covered by export restrictions for US companies and individuals:

> [exports and reexports of encryption software, like exports and reexports of encryption hardware, are controlled because of this functional capacity to encrypt information, and not because of any informational or theoretical value that such software may reflect, contain, or represent, or that its export or reexport may convey to others abroad](https://www.ecfr.gov/cgi-bin/text-idx?SID=2eaed81ac00fdd679e7c20e04882e2d6&mc=true&node=se15.2.742_115&rgn=div8#browse:~:text=exports%20and%20reexports%20of%20encryption%20software%2C,reexport%20may%20convey%20to%20others%20abroad.)

So, it isn't the fact that it is a novel algorithm, it is the fact that it can strongly encrypt data at all. The escape hatch to having to be licensed for export/reexport of such source code appears to be that they cannot limit free speech of companies or citizens, so published works are exempt (thanks largely to Phil Zimmermann, Daniel J. Bernstein, and the EFF). The trick is they have defined published as the URL being sent to those two addresses on every change of the URL hosting the code or (if changes are not available via the URL) every change of the code being sent to those two addresses.

> [(2) Notification requirement. You must notify BIS and the ENC Encryption Request Coordinator via email of the Internet location (e.g., URL or Internet address) of the publicly available encryption source code classified under ECCN 5D002 or provide each of them a copy of the publicly available encryption source code. If you update or modify the source code, you must also provide additional copies to each of them each time the cryptographic functionality of the source code is updated or modified. In addition, if you posted the source code on the Internet, you must notify BIS and the ENC Encryption Request Coordinator each time the Internet location is changed, but you are not required to notify them of updates or modifications made to the encryption source code at the previously notified location. In all instances, submit the notification or copy to crypt@bis.doc.gov and to enc@nsa.gov.](https://www.ecfr.gov/cgi-bin/text-idx?SID=2eaed81ac00fdd679e7c20e04882e2d6&mc=true&node=se15.2.742_115&rgn=div8#browse:~:text=(2)%20Notification%20requirement.%20You%20must%20notify,copy%20to%20crypt%40bis.doc.gov%20and%20to%20enc%40nsa.gov.)

So it looks like this doesn't need to be a git hook, just a one time email of the repo URL to those addresses, which is something I think we can handle on our own (if someone hasn't beaten us to it already).  It might be handy for other US companies/citizens if you mention that you have emailed the repo URL to them in the README.md (with a reference to that document), but I don't think it is needed.

**Comment by bartbutler** [comment]
Thanks for the detailed explanation. Go for it, and if feel free to make a PR to edit the README as well.

**Comment by tomholub** [comment]
Not a lawyer. The email is indeed one time only. openpgpjs is indeed subject simply by being hosted in the US (GitHub) then exported world wide. The single email fulfills the obligations as long as the algorithms are known and the code is public.

What is more interesting though @cowens to you is how your own app or code is treated under the us export laws (and other country import laws), for which you may need to ask a lawyer, since it depends on many factors (mass market or not, public or private, developers where, hosted where, submitted where and how and to whom, etc)
